### PR TITLE
Add: documentation

### PR DESCRIPTION
**Pull request by zewolfe**
This PR adds the following:
- Documentation

Issue - https://github.com/giantswarm/roadmap/issues/1254

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,129 @@
...
+Workload identity being enabled on this cluster. Note, when you enable Workload Identity on a cluster, GKE automatically creates a fixed workload identity pool for the cluster's Google Cloud project.
+A workload identity pool allows IAM to understand and trust Kubernetes service account credentials.
+The workload identity pool has the following format:
+```
+  PROJECT_ID.svc.id.goog
+```
+
+#### 2. Register your cluster
+```
+export CLUSTER_NAME="<insert-cluster-name-here>"                                                                                      at ⎈ gs-gohan-k1ng-clientcert
```
Comment (AverageMarcus) [inline README.md:33]:
```suggestion
export CLUSTER_NAME="<insert-cluster-name-here>" 
```

File: README.md [hunk 2]
```diff
@@ -1,2 +1,129 @@
...
+export MEMBERSHIP_NAME="$CLUSTER_NAME-workload-identity"
+export GCP_PROJECT_NAME="<insert-project-name-here>"
+export KUBECONFIG_CONTEXT="<insert-context-here>"
+export KUBECONFIG_PATH="~/.kube/config"
+```
+```
+gcloud container hub memberships register "$MEMBERSHIP_NAME" \                                                          at ⎈ gs-gohan-k1ng-clientcert
```
Comment (AverageMarcus) [inline README.md:40]:
```suggestion
gcloud container hub memberships register "$MEMBERSHIP_NAME" \
```

**Review by AverageMarcus — COMMENTED** [review]
LGTM but I think I'll leave the approval for someone who actually knows about the workload identity as I'm not sure.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,129 @@
...
+    --member="serviceAccount:$WORKLOAD_ID_POOL[$KUBE_NAMESPACE/$KUBE_SA_NAME]"
+```
+
+##### 3.5 Add computer viewer permissions
+```
+  # give the service account permissions to view compute resources
+  gcloud projects add-iam-policy-binding "$GCP_PROJECT_NAME" \
+    --role=roles/compute.viewer \
+    --member="serviceAccount:$GOOGLE_SA_ID"
+```
```
Comment (mnitchev) [inline README.md:95]: This isn't exactly **necessary**. I would rather say `Add necessary permissions to the GCP Service Account` and specify that these should be the permissions necessary for the workload to function. Maybe keep the `compute.viewer` as an example

**Review by mnitchev — APPROVED** [review]
Just a minor thing, otherwise looks good

File: README.md (hunk 3 above)
Comment (zewolfe) [reply README.md:95]: Good catch 👊🏾

**Pull request merged**